crates/re_query/src/archetype_view.rs

```diff
@@ -343,7 +343,9 @@ impl<A: Archetype> ArchetypeView<A> {
 
         let component = self.components.get(&C::name());
 
-        if let Some(component) = component {
+        let is_empty = component.map_or(true, |c| c.is_empty());
+
+        if let (Some(component), false) = (component, is_empty) {
```

Comment:
Please motivate this with a comment, i.e. why only non-empty components?

And does the code handle splat components (length=1) correctly?